src/front/glsl/variables.rs

```diff
@@ -19,6 +19,9 @@ impl Program {
                         return Err(ErrorKind::VariableNotAvailable(name.into()));
                     }
                 };
+                if let Some(global_var) = self.context.lookup_global_var_exps.get(name) {
```

Comment:
there is a bit of a problem here - we have this check both inside the match arms and outside of it (near the end of the function). How about the following instead: we move this check to the *beginning* of the function. Then within match arms we can safely assume that the global is not known. So we can use `append` instead of `fetch_or_append`, and we can add the expressions like we do today.

Reply:
Right, was kind of contemplating this, but wasn't quite sure how it would affect precedence. I'll try moving both this and the local_vars check up front - I think that should work well.